### ISSUE TITLE
Tighten up magic signature for /proc/pid/stat

### ISSUE THREAD

**Issue by kellyjonbrazil** [kellyjonbrazil/jc#338]
Per https://news.ycombinator.com/item?id=34093845

The `proc-pid-stat` parser seems to work fine, but the magic signature needs to be tightened up for edge cases like
```
% echo '2001 (bad) S 1 2 3 4 5) S 1888 2001 1888 34816 2001 4202496 428 0 0 0 0 0 0 0 20 0 1 0 75513 115900416 297 18446744073709551615 4194304 5100612 140737020052256 140737020050904 140096699233308 0 65536 4 65538 18446744072034584486 0 0 17 0 0 0 0 0 0 7200240 7236240 35389440 140737020057179 140737020057223 140737020057223 140737020059606 0' | jc --proc             
jc:  Error - Parser issue with proc:
             ParseError: Proc file could not be identified.
             If this is the correct parser, try setting the locale to C (LC_ALL=C).
             For details use the -d or -dd option. Use "jc -h --proc" for help.
```
Comm name `bad) S 1 2 3 4 5` is tricking the signature.

**Comment by kellyjonbrazil** [comment]
Fixed and test added. Will be merged into the next release.

**Comment by kellyjonbrazil** [comment]
Fixed in v1.22.4.
